hypervisor/debug/profiling.c

```diff
@@ -812,12 +812,12 @@ static void profiling_stop_pmu(void)
 int32_t profiling_msr_ops_all_cpus(struct acrn_vm *vm, uint64_t addr)
 {
 	uint16_t i;
+	struct profiling_msr_ops_list msr_list[CONFIG_MAX_PCPU_NUM];
 	uint16_t pcpu_nums = get_pcpu_nums();
-	struct profiling_msr_ops_list msr_list[pcpu_nums];
 
 	dev_dbg(ACRN_DBG_PROFILING, "%s: entering", __func__);
 
-	if (copy_from_gpa(vm, &msr_list, addr, sizeof(msr_list)) != 0) {
+	if (copy_from_gpa(vm, &msr_list, addr, pcpu_nums*sizeof(struct profiling_msr_ops_list)) != 0) {
```

Comment:
In this case, we assume pcpu_num <= CONFIG_MAX_PCPU_NUM.

We need to make sure this by 1) ASSERT, 2) add @post at get_pcpu_nums()

Reply:
Shuo posted a patch "https://github.com/projectacrn/acrn-hypervisor/pull/3691/commits/70d15a11e1262d304e1d0a637d208d445a127042" to change this.
I think shuo' patch should add the check "pcpu_num == CONFIG_PCPU_NUM" in init_percpu_lapic_id.
Then we could remove all the check "get_pcpu_nums() > CONFIG_MAX_PCPU_NUM" in our code and here.

Reply:
yes will guarantee phys_cpu_num <= CONFIG_MAX_PCPU_NUM in init_percpu_lapic_id, no need to check it later

Reply:
@dongyaozu have added another patch to  check phys_cpu_num in init_pcpu_pre, please help review.